sklearn/gaussian_process/kernels.py

```diff
@@ -0,0 +1,940 @@
+    """
+
+    @property
+    def theta(self):
+        return np.asarray(self.l)
+
+    @theta.setter
+    def theta(self, theta):
+        self.l = theta
+
+    def __call__(self, X, Y=None, eval_gradient=False):
+        """ Return the kernel k(X, Y) and optionally its gradient.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples_X, n_features)
+            Left argument of the returned kernel k(X, Y)
+
+        Y : array, shape (n_samples_Y, n_features), (optional, default=None)
+            Right argument of the returned kernel k(X, Y). If None, k(X, X)
+            if evaluated instead.
+
+        eval_gradient : bool (optional, default=False)
+            Determines whether the gradient with respect to the kernel
+            hyperparameter is determined. Only supported when Y is None.
+
+        Returns
+        -------
+        K : array, shape (n_samples_X, n_samples_Y)
+            Kernel k(X, Y)
+
+        K_gradient : array (opt.), shape (n_samples_X, n_samples_X, n_dims)
+            The gradient of the kernel k(X, X) with repect to the
+            hyperparameter of the kernel. Only returned when eval_gradient
+            is True.
+        """
+        if Y is None:
+            dists = pdist(X / self.l, metric='sqeuclidean')
+            K = np.exp(-.5 * dists)
+            # convert from upper-triangular matrix to square matrix
+            K = squareform(K)
+            np.fill_diagonal(K, 1)
+        else:
+            if eval_gradient:
+                raise ValueError(
+                    "Gradient can only be evaluated when Y is None.")
+            dists = cdist(X / self.l, Y / self.l, metric='sqeuclidean')
+            K = np.exp(-.5 * dists)
+
+        if eval_gradient:
+            if self.l.shape[0] == 1:
+                K_gradient = \
+                    (K * squareform(dists) / self.l)[:, :, np.newaxis]
+                return K, K_gradient
+            elif self.l.shape[0] == X.shape[1]:
+                # We need to recompute the pairwise dimension-wise distances
+                D = (X[:, np.newaxis, :] - X[np.newaxis, :, :]) ** 2 \
+                    / (self.l ** 3)
+                K_gradient = K[..., np.newaxis] * D
+                return K, K_gradient
+            else:
+                raise Exception("Anisotropic kernels require that the number "
+                                "of length scales and features match.")
+        else:
+            return K
+
+
+class RationalQuadratic(Kernel):
+    """ Rational Quadratic kernel.
+
+    This kernel can be seen as a scale mixture (an infinite sum) of RBF kernels
+    with different characteristic length-scales.
+
+    Only isotropic variant is supported at the moment.
+
+    Tunable kernel parameters
+    -------------------------
+    alpha : float > 0
+        Scale mixture parameter
+    l : float > 0
+        The length scale of the kernel.
+    """
+
+    def __init__(self, theta=[1.0, 1.0], thetaL=1e-5, thetaU=np.inf):
```

Comment:
Do not have a list as default value. It is mutable!

Why don't you call the parameters `alpha` and `l`?